src/app/models/field/emailField.ts

```diff
@@ -1,8 +1,21 @@
+import { isEmpty } from 'lodash'
 import { Schema } from 'mongoose'
 
 import { validateEmailDomains } from '../../../shared/util/email-domain-validation'
 import { IEmailFieldSchema, ResponseMode } from '../../../types'
 
+const validateAllowDomainsToBeSet = (
+  emailDomains: string[],
+  hasAllowedEmailDomains: boolean,
+): boolean => {
+  // Case 1: hasAllowedEmailDomains is false and emailDomains must be empty.
+  const case1 = !hasAllowedEmailDomains && isEmpty(emailDomains)
+  // Case 2: hasAllowedEmailDomains is true and email domains must not be empty.
+  const case2 = hasAllowedEmailDomains && !isEmpty(emailDomains)
+
+  return case1 || case2
```

Comment:
Technically due to the pre-validate hook below, this can be simplified to

```ts
return !hasAllowedEmailDomains || !isEmpty(emailDomains)
```

but idk if I want to since that relies on the other ppl looking at this code making the connection

Reply:
the existing code provides greater clarity, although I'd have used `emailDomains.length` instead of lodash's `isEmpty`